acceptancetests/assess_persistent_storage.py

```diff
@@ -120,29 +110,120 @@ def assess_charm_deploy_and_remove_single_storage(client):
             pool_storage, pool_setting))
             sys.exit(1)
+        else:
+            log.info('Check properties of single block device storage unit: PASSED.')
+
+        # check type, persistent setting and pool of single filesystem storage unit
+        storage_type = storage_output['storage'][single_fs_id]['kind']
+        persistent_setting = storage_output['storage'][single_fs_id]['persistent']
+        pool_storage = storage_output['filesystems']['0/0']['storage']
+        pool_setting = storage_output['filesystems']['0/0']['pool']
+
+        if storage_type != 'filesystem':
+            log.error(
+                'Incorrect type for single filesystem storage detected '\
+                '- {}, test terminated.'.format(storage_type))
+            sys.exit(1)
+        elif persistent_setting != False:
+            log.error(
+                'Incorrect value for persistent setting on single filesystem '\
+                'storage - {}, test terminated.'.format(persistent_setting))
+            sys.exit(1)
+        elif (pool_storage != single_fs_id) and (pool_setting != 'rootfs'):
+            log.error('Incorrect filesystems detected '\
+            '- {} with {}, test terminated.'.format(
+            pool_storage, pool_setting))
+            sys.exit(1)
+        else:
+            log.info('Check properties of single filesystem storage unit: PASSED.')
+    return (single_fs_id, single_blk_id)
 
 
+def assess_charm_removal_single_block_and_filesystem_storage(client):
+    """This function tests charm removal while a single filesystem storage
+       and a single persistent block device storage attached.
+
+       Steps taken to the test:
+       - Run assess_charm_deploy_single_block_and_filesystem_storage(client)
+       - Remove dummy-storage charm while a single block storage unit (ebs)
+         and a single filesystem storage unit (rootfs) attached.
+           > The output should states that there is a persistent storage unit.
+           > The application should be removed successfully.
+       - Check charm storage units once the removal is done.
+           > The filesystem storage unit (rootfs)should be removed successfully.
+           > The block device storage unit (ebs) should remain and detached,
+
+       :param client: ModelClient object to deploy the charm on.
+    """
+
+    charm_name = 'dummy-storage'
+    single_fs_id, single_blk_id = \
+        assess_charm_deploy_single_block_and_filesystem_storage(client)
+    # Run juju remove-application dummy-storage
+    app_removal_output = client.get_juju_output(
+        'remove-application', charm_name, '--show-log', include_e=False, merge_stderr=True)
+    ipdb.set_trace()
+    # pre-set the result to FAILED
+    remove_app_output_check = 'FAILED'
+    remove_single_fs_output_check = 'FAILED'
+    detach_single_blk_output_check = 'FAILED'
+
+    for line in app_removal_output.split('\n'):
+        if line.find('will remove unit {}'.format(charm_name)) != -1:
```

Comment:
I thought about using "in" at the beginning then realised that I have to use find because the sub-string, ('will remove unit <charm_name>') can't be separated - it would be a bug otherwise. By using "in", we first need to decide how to slice the string (line is a string here as an element of app_removal_output.split('\n') ), then whatever we do, "line" will become a list of individual word(s), but we need to check sub-string in a whole. To sum up, using find() is the easiest way to address it.